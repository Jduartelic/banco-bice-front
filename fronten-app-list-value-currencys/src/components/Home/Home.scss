@import '../../../styles/shared.scss';

.home-landing__container {
  display: flex;
  flex-direction: column;
  width: 100%;
  align-items: center;
  justify-content: center;
  height: 100%;
}

.home-landing__container-grid {
  display: grid;
  grid-template-columns: 100%;
  grid-template-rows: auto;
  width: 100%;
  row-gap: 1rem;
  height: auto;
  overflow-y: scroll;
  padding-bottom: 2rem;

  @include tablet {
    column-gap: 4%;
    grid-template-columns: 30% 68%;
  }

  @include desktop {
    grid-template-columns: 28% 74%;
  }
}

.home-landing__title {
  font: $h1-xs;
  color: $color-primary-bice;
  padding: 1rem 0;
}
