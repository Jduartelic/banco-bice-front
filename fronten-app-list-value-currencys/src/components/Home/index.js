import React, { Fragment } from 'react';
import Router from 'next/router';
import PropTypes from 'prop-types';
import ProductBox from '@components/Common/ProductBox';

import './Home.scss';

class Home extends React.Component {
  render() {
    const { commoditys } = this.props;

    return (
      <section className="home-landing__container">
        <h1 className="home-landing__title">INDEX MarketPlace</h1>

        <ul className="home-landing__container-grid">
          {Object.keys(commoditys).map((item, index) => {
            const commodity = commoditys[item];

            const productBoxProps = { commodity, uniqueVal: index };

            return <ProductBox {...productBoxProps} />;
          })}
        </ul>
      </section>
    );
  }
}

Home.propTypes = {
  commoditys: PropTypes.object.isRequired,
};

export default Home;
