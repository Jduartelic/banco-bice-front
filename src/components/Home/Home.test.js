import React from 'react';
import renderer from 'react-test-renderer';
import { configure, mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

import Home from './';

configure({ adapter: new Adapter() });

const data = {
  commoditys: {
    cobre: { key: 'cobre', name: 'Precio del Cobre, dólares por libra', unit: 'dolar', date: 1584489600, value: 2.39 },
    dolar: { key: 'dolar', name: 'Dólar observado', unit: 'pesos', date: 1584489600, value: 855.09 },
    euro: { key: 'euro', name: 'Euro', unit: 'pesos', date: 1584489600, value: 938.42 },
    ipc: {
      key: 'ipc',
      name: 'Indice de Precios al Consumidor (Var. c/r al período anterior)',
      unit: 'porcentual',
      date: 1577836800,
      value: 1.1,
    },
    ivp: { key: 'ivp', name: 'Indice de valor promedio', unit: 'pesos', date: 1586390400, value: 29706.22 },
    oro: { key: 'oro', name: 'Precio del Oro, dólares por onza', unit: 'dolar', date: 1584576000, value: 1473.2 },
    plata: {
      key: 'plata',
      name: 'Precio de la Plata, dólares por onza',
      unit: 'dolar',
      date: 1584576000,
      value: 11.69,
    },
    uf: { key: 'uf', name: 'Unidad de fomento', unit: 'pesos', date: 1586390400, value: 28630.63 },
    utm: { key: 'utm', name: 'Unidad tributaria mensual', unit: 'pesos', date: 1583020800, value: 50021 },
  },
};

describe('Home', () => {
  test('snapshot renders Home Component', () => {
    const component = renderer.create(<Home {...data} />);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('passes all props to Home and renders the ProductBoxes(li) ', () => {
    const wrapper = mount(<Home {...data} />);
    const homeWrapper = wrapper.find(Home);
    const lengthData = Object.keys(data.commoditys).length;
    expect(homeWrapper.find('li').length).toEqual(lengthData);
  });
});
