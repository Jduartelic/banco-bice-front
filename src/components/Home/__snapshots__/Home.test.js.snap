// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home snapshot renders Home Component 1`] = `
<section
  className="home-landing__container"
>
  <h1
    className="home-landing__title"
  >
    INDEX MarketPlace
  </h1>
  <ul
    className="home-landing__container-grid"
  >
    <li
      className="product-box__container"
      onClick={[Function]}
    >
      <div
        className="product-box__content-container"
      >
        <div
          className="product-box__content"
        >
          <div
            className="product-box__header-commodity"
          >
            <label
              className="product-box__brand-title"
            >
              cobre
            </label>
            <label
              className="product-box__main-title"
            >
              Precio del Cobre, dólares por libra
            </label>
          </div>
          <div
            className="product-box__content-commodity"
          >
            <label
              className="product-box__price-title"
            >
              2.39
            </label>
            <label
              className="product-box__unit-title"
            >
              / $ - USD
            </label>
          </div>
          <div
            className="product-box__footer-commodity"
          >
            <label
              className="product-box__date-title"
            >
              17/03/2020
            </label>
          </div>
        </div>
      </div>
    </li>
    <li
      className="product-box__container"
      onClick={[Function]}
    >
      <div
        className="product-box__content-container"
      >
        <div
          className="product-box__content"
        >
          <div
            className="product-box__header-commodity"
          >
            <label
              className="product-box__brand-title"
            >
              dolar
            </label>
            <label
              className="product-box__main-title"
            >
              Dólar observado
            </label>
          </div>
          <div
            className="product-box__content-commodity"
          >
            <label
              className="product-box__price-title"
            >
              855.09
            </label>
            <label
              className="product-box__unit-title"
            >
              / $ - CLP
            </label>
          </div>
          <div
            className="product-box__footer-commodity"
          >
            <label
              className="product-box__date-title"
            >
              17/03/2020
            </label>
          </div>
        </div>
      </div>
    </li>
    <li
      className="product-box__container"
      onClick={[Function]}
    >
      <div
        className="product-box__content-container"
      >
        <div
          className="product-box__content"
        >
          <div
            className="product-box__header-commodity"
          >
            <label
              className="product-box__brand-title"
            >
              euro
            </label>
            <label
              className="product-box__main-title"
            >
              Euro
            </label>
          </div>
          <div
            className="product-box__content-commodity"
          >
            <label
              className="product-box__price-title"
            >
              938.42
            </label>
            <label
              className="product-box__unit-title"
            >
              / $ - CLP
            </label>
          </div>
          <div
            className="product-box__footer-commodity"
          >
            <label
              className="product-box__date-title"
            >
              17/03/2020
            </label>
          </div>
        </div>
      </div>
    </li>
    <li
      className="product-box__container"
      onClick={[Function]}
    >
      <div
        className="product-box__content-container"
      >
        <div
          className="product-box__content"
        >
          <div
            className="product-box__header-commodity"
          >
            <label
              className="product-box__brand-title"
            >
              ipc
            </label>
            <label
              className="product-box__main-title"
            >
              Indice de Precios al Consumidor (Var. c/r al período anterior)
            </label>
          </div>
          <div
            className="product-box__content-commodity"
          >
            <label
              className="product-box__price-title"
            >
              1.1
            </label>
            <label
              className="product-box__unit-title"
            >
              / % - 
            </label>
          </div>
          <div
            className="product-box__footer-commodity"
          >
            <label
              className="product-box__date-title"
            >
              31/12/2019
            </label>
          </div>
        </div>
      </div>
    </li>
    <li
      className="product-box__container"
      onClick={[Function]}
    >
      <div
        className="product-box__content-container"
      >
        <div
          className="product-box__content"
        >
          <div
            className="product-box__header-commodity"
          >
            <label
              className="product-box__brand-title"
            >
              ivp
            </label>
            <label
              className="product-box__main-title"
            >
              Indice de valor promedio
            </label>
          </div>
          <div
            className="product-box__content-commodity"
          >
            <label
              className="product-box__price-title"
            >
              29706.22
            </label>
            <label
              className="product-box__unit-title"
            >
              / $ - CLP
            </label>
          </div>
          <div
            className="product-box__footer-commodity"
          >
            <label
              className="product-box__date-title"
            >
              08/04/2020
            </label>
          </div>
        </div>
      </div>
    </li>
    <li
      className="product-box__container"
      onClick={[Function]}
    >
      <div
        className="product-box__content-container"
      >
        <div
          className="product-box__content"
        >
          <div
            className="product-box__header-commodity"
          >
            <label
              className="product-box__brand-title"
            >
              oro
            </label>
            <label
              className="product-box__main-title"
            >
              Precio del Oro, dólares por onza
            </label>
          </div>
          <div
            className="product-box__content-commodity"
          >
            <label
              className="product-box__price-title"
            >
              1473.2
            </label>
            <label
              className="product-box__unit-title"
            >
              / $ - USD
            </label>
          </div>
          <div
            className="product-box__footer-commodity"
          >
            <label
              className="product-box__date-title"
            >
              18/03/2020
            </label>
          </div>
        </div>
      </div>
    </li>
    <li
      className="product-box__container"
      onClick={[Function]}
    >
      <div
        className="product-box__content-container"
      >
        <div
          className="product-box__content"
        >
          <div
            className="product-box__header-commodity"
          >
            <label
              className="product-box__brand-title"
            >
              plata
            </label>
            <label
              className="product-box__main-title"
            >
              Precio de la Plata, dólares por onza
            </label>
          </div>
          <div
            className="product-box__content-commodity"
          >
            <label
              className="product-box__price-title"
            >
              11.69
            </label>
            <label
              className="product-box__unit-title"
            >
              / $ - USD
            </label>
          </div>
          <div
            className="product-box__footer-commodity"
          >
            <label
              className="product-box__date-title"
            >
              18/03/2020
            </label>
          </div>
        </div>
      </div>
    </li>
    <li
      className="product-box__container"
      onClick={[Function]}
    >
      <div
        className="product-box__content-container"
      >
        <div
          className="product-box__content"
        >
          <div
            className="product-box__header-commodity"
          >
            <label
              className="product-box__brand-title"
            >
              uf
            </label>
            <label
              className="product-box__main-title"
            >
              Unidad de fomento
            </label>
          </div>
          <div
            className="product-box__content-commodity"
          >
            <label
              className="product-box__price-title"
            >
              28630.63
            </label>
            <label
              className="product-box__unit-title"
            >
              / $ - CLP
            </label>
          </div>
          <div
            className="product-box__footer-commodity"
          >
            <label
              className="product-box__date-title"
            >
              08/04/2020
            </label>
          </div>
        </div>
      </div>
    </li>
    <li
      className="product-box__container"
      onClick={[Function]}
    >
      <div
        className="product-box__content-container"
      >
        <div
          className="product-box__content"
        >
          <div
            className="product-box__header-commodity"
          >
            <label
              className="product-box__brand-title"
            >
              utm
            </label>
            <label
              className="product-box__main-title"
            >
              Unidad tributaria mensual
            </label>
          </div>
          <div
            className="product-box__content-commodity"
          >
            <label
              className="product-box__price-title"
            >
              50021
            </label>
            <label
              className="product-box__unit-title"
            >
              / $ - CLP
            </label>
          </div>
          <div
            className="product-box__footer-commodity"
          >
            <label
              className="product-box__date-title"
            >
              29/02/2020
            </label>
          </div>
        </div>
      </div>
    </li>
  </ul>
</section>
`;
